In [1]:
import pandas as pd

In [3]:
data = pd.read_html('https://www.baseball-reference.com/leagues/MLB/2018-misc.shtml')

In [13]:
df = pd.DataFrame(data[0])

Convert Est.Payroll column to integer

In [27]:
df['Payroll'] = df['Est. Payroll'].str[1:]
df['Payroll'] = df['Payroll'].str.replace(',', '')
df['Payroll'] = df['Payroll'].astype(int)
df = df.drop('Est. Payroll', axis = 1)

In [38]:
df.head()

Tm  Attendance  Attend/G  BatAge  PAge  BPF  PPF  #HOF  #A-S  #a-tA-S  \
0  ARI     2242695     27688    29.2  29.6  105  104     0     3       11   
1  ATL     2555781     31553    27.4  27.7  102  101     0     4       13   
2  BAL     1564192     19311    28.4  27.1   98  100     0     1       11   
3  BOS     2895575     35748    27.7  28.9  105  103     0     5       16   
4  CHC     3181089     38794    27.2  30.2  104  103     0     3       14   

   Time  Chall  Succ Succ%   Managers    Payroll  
0  3:10     45    26   57%    Lovullo  134850600  
1  3:04     45    18   40%    Snitker  115848667  
2  3:01     27    15   55%  Showalter  141555833  
3  3:13     43    22   51%       Cora  222205000  
4  3:09     43    27   62%     Maddon  205373881

In [48]:
df.sort_values('Attend/G', ascending=False)

Tm  Attendance  Attend/G  BatAge  PAge  BPF  PPF  #HOF  #A-S  #a-tA-S  \
13  LAD     3857500     47043    28.1  29.2   97   95     0     3       18   
18  NYY     3482855     42998    27.2  28.8  100   98     0     4       16   
25  STL     3403587     42020    28.1  26.7   97   96     0     2       12   
24  SFG     3156185     38965    29.8  28.6   93   94     0     2       14   
4   CHC     3181089     38794    27.2  30.2  104  103     0     3       14   
12  LAA     3020216     37287    29.5  27.5   98   98     0     1        7   
8   COL     3015880     37233    28.7  27.9  117  117     0     3        9   
10  HOU     2980549     36797    28.1  30.0   99   96     0     6       14   
3   BOS     2895575     35748    27.7  28.9  105  103     0     5       16   
15  MIL     2850875     35196    28.9  28.8  102  102     0     5       14   
1   ATL     2555781     31553    27.4  27.7  102  101     0     4       13   
29  WSN     2529604     31230    27.5  30.2  106  105     0     2       14   
28  TOR     2325281     28707    29.0  29.3   98   99     0     1       11   
23  SEA     2299489     28389    29.8  29.0   95   95     0     4       13   
0   ARI     2242695     27688    29.2  29.6  105  104     0     3       11   
17  NYM     2224995     27469    28.4  28.0   92   92     0     1       17   
22  SDP     2168536     26772    26.9  27.7   93   94     0     1        6   
20  PHI     2158124     26644    26.7  26.8  102  102     0     1        8   
27  TEX     2107107     26014    27.4  30.4  110  111     0     1        9   
16  MIN     1959197     24188    28.2  28.5  100  100     0     1       13   
7   CLE     1926701     23786    29.4  29.4  103  102     0     6       15   
9   DET     1856970     22926    27.9  28.6  103  105     0     1        9   
11  KCR     1665107     20557    28.6  27.6  100  101     0     1        6   
6   CIN     1629356     20116    27.2  27.1  102  102     0     3        8   
5   CHW     1608817     19862    26.5  27.6   98   99     0     1        6   
19  OAK     1573616     19427    28.0  29.2   96   96     0     2       10   
2   BAL     1564192     19311    28.4  27.1   98  100     0     1       11   
21  PIT     1465316     18316    27.8  26.5   97   98     0     1        8   
26  TBR     1154973     14259    27.1  27.1   97   96     0     2        9   
14  MIA      811104     10014    27.5  27.6   94   95     0     1        4   

    Time  Chall  Succ Succ%                Managers    Payroll  
13  3:13     39    18   46%                 Roberts  164703429  
18  3:11     35    26   74%                   Boone  160743032  
25  3:10     47    18   38%      Matheny and Shildt  157713667  
24  3:01     31    23   74%                   Bochy  202060277  
4   3:09     43    27   62%                  Maddon  205373881  
12  3:09     42    23   54%                Scioscia  166849666  
8   3:04     46    24   52%                   Black  136658500  
10  3:04     34    14   41%                   Hinch  172781200  
3   3:13     43    22   51%                    Cora  222205000  
15  3:07     34     7   20%                Counsell  109295700  
1   3:04     45    18   40%                 Snitker  115848667  
29  3:07     40    21   52%                Martinez  188886699  
28  2:59     39    18   46%                 Gibbons  151670772  
23  2:58     37    13   35%                 Servais  157090065  
0   3:10     45    26   57%                 Lovullo  134850600  
17  3:01     29    15   51%                Callaway  161403844  
22  3:00     44    19   43%                   Green   93821067  
20  3:10     44    27   61%                  Kapler   93874333  
27  3:04     42    17   40%  Banister and Wakamatsu  106099628  
16  3:10     45    21   46%                 Molitor  110275000  
7   3:03     37    16   43%                Francona  143375233  
9   3:00     32    19   59%              Gardenhire  111531000  
11  2:57     41    31   75%                    Yost   95199167  
6   3:03     38    14   36

In [47]:
pd.read_html('https://www.baseball-reference.com/leagues/MLB/2018-standings.shtml')

[                  Tm    W    L   W-L%    GB
 0     Boston Red Sox  108   54  0.667    --
 1   New York Yankees  100   62  0.617   8.0
 2     Tampa Bay Rays   90   72  0.556  18.0
 3  Toronto Blue Jays   73   89  0.451  35.0
 4  Baltimore Orioles   47  115  0.290  61.0,
                    Tm   W    L   W-L%    GB
 0   Cleveland Indians  91   71  0.562    --
 1     Minnesota Twins  78   84  0.481  13.0
 2      Detroit Tigers  64   98  0.395  27.0
 3   Chicago White Sox  62  100  0.383  29.0
 4  Kansas City Royals  58  104  0.358  33.0,
                    Tm    W   L   W-L%    GB
 0      Houston Astros  103  59  0.636    --
 1   Oakland Athletics   97  65  0.599   6.0
 2    Seattle Mariners   89  73  0.549  14.0
 3  Los Angeles Angels   80  82  0.494  23.0
 4       Texas Rangers   67  95  0.414  36.0,
                       Tm   W   L   W-L%    GB
 0         Atlanta Braves  90  72  0.556    --
 1   Washington Nationals  82  80  0.506   8.0
 2  Philadelphia Phillies  80  82  0.494  10.0

<h1>MLB STATS API PLAYGROUND

In [50]:
import statsapi

AL Standings

<h2>Rookie HR Leaders

In [158]:
rookie_hr_leaders = statsapi.league_leaders('homeRuns',season=2019, playerPool = 'rookies', limit=5)
rookie_hr_leaders = rookie_hr_leaders.split('\n')
rookie_hr_leaders

['Rank Name                 Team                    Value',
 ' 1   Pete Alonso          New York Mets            49  ',
 ' 2   Eloy Jimenez         Chicago White Sox        28  ',
 ' 3   Yordan Alvarez       Houston Astros           26  ',
 ' 3   Christian Walker     Arizona Diamondbacks     26  ',
 ' 5   Fernando Tatis Jr.   San Diego Padres         22  ',
 '']

<h1>GET DAILY STANDINGS

Understanding the syntax:<br><br>
LEAGUES<br>
103 = AL <br>
104 = NL <br><br>
DIVISIONS<br>
200 = AL West<br>
201 = AL East<br>
202 = AL Central<br>
203 = NL West<br>
204 = NL East<br>
205 = NL Central<br>

In [156]:
def get_division_standings(league, division):
    standings = pd.DataFrame(statsapi.standings_data(league)[division]['teams'])
    standings = standings[['name', 'div_rank', 'gb', 'w', 'l', 'elim_num', 'wc_rank', 'wc_gb', 'wc_elim_num']]
    return standings    

get_division_standings(103, 201)

name div_rank    gb   w    l elim_num wc_rank wc_gb  \
0   New York Yankees        1     -  99   54        -       -     -   
1     Tampa Bay Rays        2   9.0  90   63        1       2     -   
2     Boston Red Sox        3  19.0  79   72        E       4  10.0   
3  Toronto Blue Jays        4  37.5  61   91        E       9  28.5   
4  Baltimore Orioles        5  49.5  49  103        E      11  40.5   

  wc_elim_num  
0           -  
1           -  
2           1  
3           E  
4           E